guides/stable/3D_interactivity.md

```diff
@@ -0,0 +1,131 @@
+# 3D interactivity
+
+## Coordinate systems in napari
+In napari, there are three main coordinate systems: (1) canvas, (2) world, and (3) layer. The canvas coordinates system is the 2D coordinate system of the canvas on which the scene is rendered. World coordinates are the nD coordinates of the entire scene. As the name suggests, the layer coordinates is the nD coordinate system of the data in a given layer. Layer coordinates are specific the each layer's data and are related to the world coordinate system via the layer transforms.
+
+![coordinate-systems](images/3d_interaction_coordianates.png)
+
+## In 3D mode, clicks are lines
+Since the 3D scene is rendered on a 2D surface (your screen), your mouse click does not map to a specific point in space. Thus, when rendering a scene in 3D, napari interprets your click as a line in the direction of the camera view that goes through the point that view ray intersects the camera plane.
+
+![click-line](images/3d_interaction_click_line.png)
+
+When a user clicks or moves the mouse in the canvas, napari emits a mouse event with the following properties:
+
+
+- pos: the position of the click in canvas coordinates.
+- position: the position of the click in world coordinates.
+- view_direction: a unit vector giving the direction of the camera in
+    world coordinates.
+- dims_displayed: a list of the dimensions currently being displayed
+    in the viewer. This comes from viewer.dims.displayed.
+- dims_point: the indices for the data in view in world coordinates.
+
+
+
+## Determining where the click intersects the data
+Each napari layer has a method called `get_ray_intersections()` that will return the points on the data bounding box that a given line will intersect (`start_point ` and `end_point `). `start_point` and `end_point` are  the end points of line segment that intersects the layer's axis-alinged data bounding box. `near_point` is the end point that is closest to the camera (i.e, the "first" intersection") and `far_point` is the end point that is farthest from the camera (i.e., the "last" intersection). You can use the line segment between `start_point` and `end_point` to interrogate the layer data that is "under" your cursor.
+
+![click-intersection](images/3d_interaction_ray_intersection.png)
+
+See the `get_ray_intersection()` docstrings below for details. Note that if the line does not intersect the data bounding box (i.e., the click was outside of the data), `start_point` and `end_point` are `None`. The `position` and `view_direction` should be provided as world coordinates if `world` is set to True and in layer coordinates if `world` is set to `False`.
+
+```
+def get_ray_intersections(
+    self,
+    position: List[float],
+    view_direction: np.ndarray,
+    dims_displayed: List[int],
+    world: bool = True,
+) -> Union[Tuple[np.ndarray, np.ndarray], Tuple[None, None]]:
```

Comment:
I am open to adding a convenience method or function that takes the mousevent. I think it is important to keep a public function that takes the keyword arguments above. That way if a user wants to interrogate their data outside of a mouse event callback, they don't have to create a mock `Event`.

Reply:
cool I'll try to get this done later today. Are you okay with having that function be public but tucked away in `napari.layers.utils` or somewhere similar?

Reply:
I think it would be nice to have the one that takes the event also on the layer. That way the user doesn't have to import anything to create a mouse event callback.

Reply:
slight misunderstanding I think, what I'm proposing is
- more complex utility function in some utils file (but public)
- simpler function which takes a mouse event directly on the layer

Reply:
Got it. Is there a reason not to have both on the layer? The function that takes the event basically becomes a passthrough. I think we'd want to leave at least `layer.get_value()` as is since that has been part of the public API for a while.

Reply:
no specific reason - happy for both to be on the layer with the `_from_mouse_event` nomenclature!

Reply:
Cool, let's have them both on the layer then. Thanks!